### ISSUE TITLE
Model Interface

### ISSUE THREAD

**Issue by heindorf** [dice-group/Ontolearn#28]
Currently, the model interface look like

```
model = CELOE(knowledge_base=kb,...)
model.predict(pos=p, neg=n)
model.show_best_predictions(top_n=10)
```

I would suggest the following to make it more consistent with the widely used sciki-learn library:
```
model = CELOE(...)
model.fit(X=kb, y={positives: p, negatives: n})
model.best_hypotheses(top_n=10)
model.predict(kb, hypothesis=0)
```

To summarize the changes:
  - The data `X` (kb) is provided as part of the `fit` method (and not as a hyperparameter)
  - The labels `y` are kept more general (later on we will support different learning problems, e.g., positive only)
  - The word "show" in `show_best_predictions` is omitted because the result will be returned as a list (and not printed to the terminal)
  - The word "predictions" is renamed to "hypotheses" (hypotheses is more general, even more general then "class expression")
  - The method `fit` learns the class expressions from the ground truth (not the method `predict`) and returns a **model**. This would allow a sequence of calls like `model.fit(...).predict(...)`
  - `predict`: performs the actual predictions, i.e., it evaluates a given hypothesis for every individual in the KB (the default hypothesis should be the best hypothesis)

**Comment by SimonBin** [comment]
we made some progress/changes but should still further push this when time permits
